### PR TITLE
chore: Split files for Module Instances in IDE

### PR DESCRIPTION
**Pull request by hetunandu**
As part of updating the IDE IA Redesign, we are splitting certain functionalities so that they can be extended on the EE side. 

Summary of changes:
- Create a `WithAddView` interface in RouteBuilder to indicate a scenario where the url can be suffixed with a `/add` to have edit and add views together
- Create a common GroupedList component
- Adds a dummy JS Add screen and routes. Wont navigate to it via the system
- Move `EntityItem` interface to a more common location for extensibility
- Update FocusElement config type to allow for a limited targeting in FocusStrategy
- Extracted out ListItem render components for JS and Query
- Moved JS and Query segment hooks to a CE location for extensibility of AddQuery and ListQuery
- Extracted EditorPaneRoutes for extensibility
- Added FocusEntity entries for QueryModuleInstance and JSModuleInstance. This will be cleaned up at a later date.

Companion EE PR: https://github.com/appsmithorg/appsmith-ee/pull/3389

Fixes: #30544

## Summary by CodeRabbit

- **New Features**
  - Introduced an interface for adding views to enhance URL generation for different editor functions.
  - Added new user-facing messages for creating blank JavaScript objects in the editor.
  - Implemented new hooks and components to manage and render JavaScript and Query entities within the IDE.
  - Created a grouped list component for categorizing items within the editor pane.

- **Enhancements**
  - Extended existing enums and utilities to support new module instance types and actions.
  - Improved navigation and focus management within the IDE based on user interaction and context.
  - Streamlined the process of adding JavaScript operations and queries in the editor.

- **Refactor**
  - Updated import paths for better code organization and module resolution.
  - Removed redundant code and replaced components with more efficient implementations.

- **Bug Fixes**
  - Corrected the behavior of the `Files` component to handle undefined types appropriately.

### CODE REVIEW

File: app/client/src/ce/constants/messages.ts [hunk 1]
```diff
@@ -2292,9 +2292,11 @@ export const EDITOR_PANE_TEXTS = {
     "Drag & drop widgets to create your app",
   query_add_button: () => "New query / API",
   js_add_button: () => "New JS object",
+  js_blank_object_item: () => "Blank JS Object",
```
[the remark this answers is not in the document]
Comment (ankitakinger) [reply app/client/src/ce/constants/messages.ts:2295]: Can the o for the object be small? Below text has it small. Can we use same casing all over new messages that we create?

File: app/client/src/pages/Editor/Explorer/Files/index.tsx [hunk 2]
```diff
@@ -109,7 +109,6 @@ function Files() {
               key={entity.id}
               searchKeyword={""}
               step={2}
-              type={type}
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/Explorer/Files/index.tsx:112]: why did we remove the type here?

**Review by ankitakinger — APPROVED** [review]

File: app/client/src/ce/constants/messages.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (hetunandu) [reply app/client/src/ce/constants/messages.ts:2295]: Missed this one. Thanks

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**